.github/workflows/release-gitjob.yaml

```diff
@@ -0,0 +1,89 @@
+# gitjob release workflow
+name: gitjob release
+
+on:
+  push:
+    tags:
+      - v**
+
+permissions:
+  contents: write
+  packages: write
+  id-token: write # this is important, it's how we authenticate with Vault
+
+env:
+  GOARCH: amd64
+  CGO_ENABLED: 0
+  SETUP_K3D_VERSION: 'v5.5.1'
+
+jobs:
+  build-gitjob:
+    runs-on: ubuntu-latest
+    if: github.repository == 'rancher/gitjob'
+    steps:
+      - name: Check out Gitjob
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+          check-latest: true
+
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v6.0.1
+        with:
+          # Required: the version of golangci-lint is required and must be specified without patch version: we always
+          # use the latest patch version.
+          version: v1.58
+          args: --timeout=10m --config=.golangci.json
+
+      - name: "Read Vault Secrets"
+        uses: rancher-eio/read-vault-secrets@main
+        with:
+          secrets: |
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials username | DOCKER_USERNAME ;
```

Comment:
Does this involve org-wide secrets, or do we need to set or check secrets on this repo?